src/user/user.service.ts

```diff
@@ -10,48 +11,78 @@ import * as dotenv from 'dotenv';
 import { LogService } from '../log/log.service';
 import { ChangeLoginDto } from './dto/change-login.dto';
 import { UploadImageDto } from './dto/upload-image.dto';
-import { includes } from 'lodash';
-import { Team } from 'src/team/team.model';
+import { Team } from '../team/team.model';
 import { BanDto } from './dto/ban.dto';
-import { where } from 'sequelize/types';
-dotenv.config();
+import * as env from 'env-var';
+dotenv.config({path: `.${env.get('NODE_ENV').required().asString()}.env`});
 
 @Injectable()
 export class UserService {
   constructor(
     @InjectModel(User) private userRepository: typeof User,
-    private roleService: RoleService, private jwtService: JwtService, private logService: LogService
+    private jwtService: JwtService, private logService: LogService
   ) {}
 
   async createUser(dto: CreateUserDto) {
-    const role = await this.roleService.getRoleByValue(ROLE[ROLE.player]);
+    let role;
+    if(dto.registered) {
+      role = ROLE.PLAYER;
+    }
+    else {
+      role = ROLE.MANAGER;
+    }
     if(!role) {
-      throw new HttpException(ROLE_USER_NOT_FOUND, HttpStatus.NOT_FOUND); 
+      //log to mongo
+      const log = {
+        message: Response.ROLE_USER_NOT_FOUND.message,
+        where: 'user.servise.ts (createUser())',
+        type: LogType.ERROR
+      }
+      await this.logService.create(log);
+
+      throw new HttpException(Response.ROLE_USER_NOT_FOUND, HttpStatus.NOT_FOUND); 
     }
-    const dtoWithRole = { ...dto, roleId: role.id, ban: false, banReason: '' };
+    const dtoWithRole = { ...dto, roleId: role, ban: false, banReason: '' };
 
     // log to mongo
-    const message = LOG_USER_CREATE.message.concat(`${dto.email}`);
+    const message = Response.LOG_USER_CREATE.message.concat(`${dto.email}`);
     const log = {
       message: message,
-      where: 'user.servise.ts',
-      type: 'create'
+      where: 'user.servise.ts (createUser())',
+      type: LogType.CREATE
     }
-    this.logService.create(log);
+    await this.logService.create(log);
     
     return await this.userRepository.create(dtoWithRole);
   }
 
   async getUserByEmail(email: string) {
-    return await this.userRepository.findOne({ where: { email } });
+    return await this.userRepository.findOne({ where: { email }, include: {all: true} });
```

Comment:
`include: {all: true}` - what's that for?

Reply:
this also finds data associated with foreign keys